Library/Formula/ice.rb

```diff
@@ -8,7 +8,20 @@ class Ice < Formula
   option 'doc', 'Install documentation'
   option 'demo', 'Build demos'
 
-  depends_on 'berkeley-db'
+  resource "berkeley-db" do
+    url 'http://download.oracle.com/berkeley-db/db-5.3.28.tar.gz'
+    sha1 'fa3f8a41ad5101f43d08bc0efb6241c9b6fc1ae9'
+
+  end
+
+  # Fix build under Xcode 4.6
+  # Double-underscore names are reserved, and __atomic_compare_exchange is now
+  # a built-in, so rename this to something non-conflicting.
+  resource "berkeley-db-patch" do
```

Comment:
I forgot about the patch here. Patching resources is something Homebrew still struggles with a bit.. I don't know if @jacknagel made a decision about how Homebrew should handle resource patching or whether it's still being discussed. Perhaps we could look at ` inreplacing ` if the maintainers aren't ready to support resource patching yet.